pkg/common/spinner.go

```diff
@@ -55,7 +56,16 @@ func NewSpinner(w io.Writer) *Spinner {
 	}
 }
 
+func isCIEnvironment() bool {
+	return os.Getenv("CI") != "" || os.Getenv("GITHUB_ACTIONS") != ""
```

Comment:
is "CI" a standard environment variable?

if not, we should be using some sort of project specific prefix, to avoid conflicting with some unrelated meaning.

there are _standard_ environment variables about the capability of the terminal, see the linked code in https://github.com/kubernetes-sigs/hydrophone/issues/210

Reply:
commented also on #210

Reply:
> is "CI" a standard environment variable?
> 
> if not, we should be using some sort of project specific prefix, to avoid conflicting with some unrelated meaning.
> 
> there are _standard_ environment variables about the capability of the terminal, see the linked code in https://github.com/kubernetes-sigs/hydrophone/issues/210

The thing is that the CI also detects the ci environment as I linked one tool in which I use the same way there and it works totally good

Reply:
I don't think we should invent our own unprefixed environment variables, this is a global namespace.

Reply:
> I don't think we should invent our own unprefixed environment variables, this is a global namespace.

Can you just tell me where I can find environment variables to identify ci in Hydrophone?

Reply:
I think a more universal approach would be to identify if we're running in a terminal or not. If not, lets disable the spinner. `x/term` has a `IsTerminal` method that we may be able to make use of. I've never used it directly but this example code seems to be promising from a few minutes of testing. https://go.dev/play/p/8aWZKSoxZE5. If we return from `Spinner.Start` if `IsTerminal` is true that may be a good enough stop gap.

Reply:
Yeah, we check for tty in https://github.com/kubernetes-sigs/kind/blob/main/pkg/internal/env/term.go

there is not a universal environment variable for "is CI" and we shouldn't attempt to invent it here

Reply:
Some CI use a fake tty, and there are other user signals that they would like to disable this, each check is commented above.

Reply:
Hey @rjsadow @BenTheElder I have modified the spinner function and used **x/term** and **IsTerminal** to check if running in non terminal environment the spinner function should not start